colossalai/engine/ophooks/__init__.py

```diff
@@ -73,6 +73,10 @@ def register_ophooks_recursively(module: torch.nn.Module, ophook_list: List[Base
     if len(list(module.parameters(recurse=False))) == 0:
         return
 
+    # Unsharded module do not need hooks
+    if not module.param_is_sharded:
```

Comment:
Wrong. This is a common utils. We can not assume it is only used for shared tensors...

Reply:
done